### PR TITLE
NFS Export operations support for NFS for powerflex v4.0

### PR DESCRIPTION
**Pull request by KshitijaKakde**
# Description
This PR includes changes for adding operation for NFS export for PowerFlex storage system v4.0

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| dell/csm#763

# Checklist:

- [X] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [X] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your 

Integration Test:
inti

Unit Test

### CODE REVIEW

**Review by khareRajshree — CHANGES_REQUESTED** [review]
@KshitijaKakde, please rebase this branch with main as I see a couple of changes not reflexting here since the last merge.

Also fix the linting errors reported from github workflow.

File: inttests/nfs_export_test.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	if os.Getenv("GOSCALEIO_NFSEXPORT") != "" {
+		return os.Getenv("GOSCALEIO_NFSEXPORT")
+	}
+
+	nfsexport, _ := C.GetNFSExport()
+	assert.NotNil(t, nfsexport)
+	if nfsexport == nil {
+		return ""
+	}
+	fmt.Printf("filesystems[0].Name: %v", nfsexport[0].Name)
```
Comment (khareRajshree) [inline inttests/nfs_export_test.go:34]: Please remove this print statement from integrations test.

File: inttests/nfs_export_test.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+	return nfsexport[0].Name
+}
+
+// func TestGetFileSystemByName(t *testing.T) {
```
Comment (khareRajshree) [inline inttests/nfs_export_test.go:38]: Please remove the commented code.

File: inttests/nfs_export_test.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+// 	if len(fsName) > 0 {
+// 		fs, err := system.GetFileSystemByName(fsName)
+// 		assert.Nil(t, err)
+// 		assert.Equal(t, fsName, fs.Name)
+// 	}
+// }
+
+func TestNFSExportByName(t *testing.T) {
+	// system := getSystem()
```
Comment (khareRajshree) [inline inttests/nfs_export_test.go:53]: Please remove the commented code.

File: inttests/nfs_export_test.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	assert.NotNil(t, err)
+	assert.Nil(t, fs)
+}
+
+func TestCreateModifyDeleteNFSExport(t *testing.T) {
+	system := getSystem()
+	assert.NotNil(t, system)
+
+	nfsName := fmt.Sprintf("%s-%s", "NFS", testPrefix+randString(8))
+	nfsmodify := "NFS export modify testing"
```
Comment (khareRajshree) [inline inttests/nfs_export_test.go:99]: Please remove this print statement from integrations test.

File: nfs_export.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	path := fmt.Sprintf("/rest/v1/nfs-exports")
+
+	var body *types.NFSExportCreate = createParams
+	err = c.getJSONWithRetry(http.MethodPost, path, body, &respnfs)
+	if err != nil {
+		return nil, err
+	}
+
+	return respnfs, nil
+
```
Comment (khareRajshree) [inline nfs_export.go:49]: Please remove this new line..

**Review by khareRajshree — CHANGES_REQUESTED** [review]
@KshitijaKakde Please address the below review comments. Thanks.

File: inttests/GOSCALEIO_TEST.env_example [hunk 6]
```diff
@@ -14,6 +14,10 @@ GOSCALEIO_INSTALLATIONID=08bhsd7f9cfy890g
 GOSCALEIO_NASSERVER=env8nasserver
 GOSCALEIO_FILESYSTEM=env8FS
 
+# For NFSExport operations , enable these values
+GOSCALEIO_NFSEXPORT=share1
+GOSCALEIO_FILESYSTEM_NFSEXPORT=twee-fs11
```
Comment (khareRajshree) [inline inttests/GOSCALEIO_TEST.env_example:19]: please put a generic dummy value for GOSCALEIO_FILESYSTEM_NFSEXPORT

**Review by khareRajshree — CHANGES_REQUESTED** [review]
for the UTs and in-tests, please update with the latest results covering for all methods.

**Review by ashleyvjoy — APPROVED** [review]

**Review by khareRajshree — APPROVED** [review]

File: types/v1/types.go [hunk 7]
```diff
@@ -1203,3 +1203,62 @@ type FsCreate struct {
...
+}
+
+// CreateResponse defines struct for response
+type CreateResponse struct {
+	ID string `json:"id,omitempty"`
+}
+
+// NFSExportCreate defines struct for Create NFS Export
+type NFSExportCreate struct {
+	Name         string `json:"name"`
```
Comment (VamsiSiddu-7) [inline types/v1/types.go:1240]: add the required field to the name id and path

File: types/v1/types.go [hunk 8]
```diff
@@ -1203,3 +1203,62 @@ type FsCreate struct {
...
+	DefaultAccess      NFSExportDefaultAccessEnum `json:"default_access,omitempty"`
+	Path               string                     `json:"path,omitempty"`
+	ReadWriteHosts     []string                   `json:"read_write_hosts,omitempty"`
+	ReadOnlyHosts      []string                   `json:"read_only_hosts,omitempty"`
+	ReadWriteRootHosts []string                   `json:"read_write_root_hosts,omitempty"`
+	ReadOnlyRootHosts  []string                   `json:"read_only_root_hosts,omitempty"`
+}
+
+// CreateResponse defines struct for response
+type CreateResponse struct {
```
Comment (VamsiSiddu-7) [inline types/v1/types.go:1234]: can you rename this to NFSExportCreateResponse

File: types/v1/types.go [hunk 9]
```diff
@@ -1203,3 +1203,62 @@ type FsCreate struct {
...
+	ReadOnly     NFSExportDefaultAccessEnum = "Read_Only"
+	ReadWrite    NFSExportDefaultAccessEnum = "Read_Write"
+	Root         NFSExportDefaultAccessEnum = "Root"
+	ReadOnlyRoot NFSExportDefaultAccessEnum = "Read_Only_Root "
+)
+
+// NFSExport defines the struct for NFSExport
+type NFSExport struct {
+	ID                 string
+	FileSystemID       string                     `json:"file_system_id,omitempty"`
```
Comment (VamsiSiddu-7) [inline types/v1/types.go:1222]: if you dont provide the `json:` tag it cannot parse the id.

**Review by VamsiSiddu-7 — CHANGES_REQUESTED** [review]
@KshitijaKakde I have added some review comments. Please address those.

**Pull request merged**